moveit_ros/perception/mesh_filter/src/depth_self_filter_nodelet.cpp

```diff
@@ -174,25 +176,28 @@ void mesh_filter::DepthSelfFiltering::filter(const sensor_msgs::ImageConstPtr& d
 
 void mesh_filter::DepthSelfFiltering::addMeshes(MeshFilter<StereoCameraModel>& mesh_filter)
 {
-  robot_model_loader::RobotModelLoader robotModelLoader("robot_description");
-  moveit::core::RobotModelConstPtr robotModel = robotModelLoader.getModel();
-  const vector<moveit::core::LinkModel*>& links = robotModel->getLinkModelsWithCollisionGeometry();
-  for (size_t i = 0; i < links.size(); ++i)
+  robot_model_loader::RobotModelLoader robot_model_loader("robot_description");
+  moveit::core::RobotModelConstPtr robot_model = robot_model_loader.getModel();
+  const auto& links = robot_model->getLinkModelsWithCollisionGeometry();
+  for (auto link : links)
   {
-    shapes::ShapeConstPtr shape = links[i]->getShape();
-    if (shape->type == shapes::MESH)
+    const auto& shapes = link->getShapes();
+    for (const auto& shape : shapes)
     {
-      const shapes::Mesh& m = static_cast<const shapes::Mesh&>(*shape);
-      MeshHandle mesh_handle = mesh_filter.addMesh(m);
-      transform_provider_.addHandle(mesh_handle, links[i]->getName());
+      if (shape->type == shapes::MESH)
+      {
+        const shapes::Mesh& m = static_cast<const shapes::Mesh&>(*shape);
+        MeshHandle mesh_handle = mesh_filter.addMesh(m);
+        transform_provider_.addHandle(mesh_handle, link->getName());
+      }
     }
   }
 }
 
 // Handles (un)subscribing when clients (un)subscribe
 void mesh_filter::DepthSelfFiltering::connectCb()
 {
-  lock_guard<mutex> lock(connect_mutex_);
+  boost::lock_guard<boost::mutex> lock(connect_mutex_);
```

Comment:
Is there a reason you are using a boost mutex and not the one from the stl?  Is it needed somewhere that takes only the boost type?